mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,376 @@
-.. include:: /../common/stub-topic.txt
-
-|stub-icon| Authoring Tag Helpers
+Authoring Tag Helpers
 ===========================================
 
-.. include:: /../common/stub-notice.txt
+By `Rick Anderson`_
+
+    - `Getting started with Tag Helpers`_
+    - `Starting the email tag helper`_
+    - `A working email tag helper`_
+    - `The bold tag helper`_
+    - `Web site information tag helper`_
+    - `Condition tag helper`_
+    - `Avoiding tag helper conflicts`_
+    - `Inspecting and retrieving child content`_
+    - `Wrap up and next steps`_
+
+You can browse the source code for the sample app used in this document on `GitHub <https://github.com/aspnet/Docs/tree/master/mvc/views/tag-helpers/authoring/sample>`__. 
+
+Getting started with tag helpers
+------------------------------------
+
+This tutorial provides an introduction to programming tag helpers. :doc:`intro` describes the benefits that tag helpers provide.
+
+A tag helper is any class that implements the `ITagHelper <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITagHelper.cs>`__ interface. However, when you author a tag helper, you generally derive from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__ because doing so gives you access to the methods and properties that are required. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial.
```

Comment:
However, when you author a tag helper, you generally derive from TagHelper<https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>, doing so gives you access to the Process method.

From: N. Taylor Mullen [mailto:notifications@github.com]
Sent: Wednesday, September 2, 2015 2:43 PM
To: aspnet/Docs <Docs@noreply.github.com>
Cc: Rick Anderson <Rick.Anderson@microsoft.com>
Subject: Re: [Docs] new TH authoring article (#419)


In mvc/views/tag-helpers/authoring.rst<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2faspnet%2fDocs%2fpull%2f419%23discussion_r38588433&data=01%7c01%7cRick.Anderson%40microsoft.com%7cd5b4095b9649499895b008d2b3df710f%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=WgWzc0a0DWEwBKT5CaWbYhwKE2tztvm%2fETHbplTJ0jc%3d>:

> +    - `A working email tag helper`_

> +    - `The bold tag helper`_

> +    - `Web site information tag helper`_

> +    - `Condition tag helper`_

> +    - `Avoiding tag helper conflicts`_

> +    - `Inspecting and retrieving child content`_

> +    - `Wrap up and next steps`_

> +

> +You can browse the source code for the sample app used in this document on `GitHub <https://github.com/aspnet/Docs/tree/master/mvc/views/tag-helpers/authoring/sample>`__.

> +

> +Getting started with tag helpers

> +------------------------------------

> +

> +This tutorial provides an introduction to programming tag helpers. :doc:`intro` describes the benefits that tag helpers provide.

> +

> +A tag helper is any class that implements the `ITagHelper <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITagHelper.cs>`__ interface. However, when you author a tag helper, you generally derive from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__ because doing so gives you access to the methods and properties that are required. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial.

because doing so gives you access to the methods and properties that are required. Nope, just adds a convenient unsynchronous Process method.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2faspnet%2fDocs%2fpull%2f419%2ffiles%23r38588433&data=01%7c01%7cRick.Anderson%40microsoft.com%7cd5b4095b9649499895b008d2b3df710f%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=5lqaawfvf%2bFt6BBPKB5ZPcwKWnOPyU7SOQ05qo4CHJM%3d>.